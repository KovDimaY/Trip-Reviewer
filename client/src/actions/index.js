import axios from 'axios';

import {
    GET_TRIPS,
    GET_USERS,
    GET_TRIP_W_REVIEWER,
    CLEAR_TRIP_W_REVIEWER,
    USER_REGISTER,
    USER_LOGIN,
    RESET_PASSWORD,
    USER_AUTH,
    ADD_TRIP,
    UPDATE_TRIP,
    DELETE_TRIP,
    GET_TRIP,
    CLEAR_NEW_TRIP,
    CLEAR_TRIP,
    GET_USER_REVIEWS
} from '../constants/action-names';

export function getTrips(
    limit = 10,
    start = 0,
    order = 'asc',
    list = []
) {
    const request = axios
        .get(`/api/getManyTrips?limit=${limit}&skip=${start}&order=${order}`);

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: GET_TRIPS,
                payload: [...list, ...response.data]
            });
        });
    }
}

export function getTripWithReviewer(id) {
    const request = axios.get(`/api/getTrip?id=${id}`);

    return (dispatch) => {
        return request.then(({ data }) => {
            const trip = data;

            return axios
                .get(`/api/getReviewer?id=${trip.ownerId}`)
                .then(({ data }) => {
                    const response = {
                        trip,
                        reviewer: data
                    };

                    dispatch({
                        type: GET_TRIP_W_REVIEWER,
                        payload: response
                    });
                });
        });
    };
}

export function getUserReviews(userId) {
    const request = axios
        .get(`/api/getUserReviews?user=${userId}`);

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: GET_USER_REVIEWS,
                payload: response.data
            });
        });
    }
}

export function getTrip(id) {
    const request = axios.get(`/api/getTrip?id=${id}`);

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: GET_TRIP,
                payload: response.data
            });
        });
    }
}

export function updateTrip(data) {
    const request = axios.post(`/api/tripUpdate`, data);

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: UPDATE_TRIP,
                payload: response.data
            });
        });
    }
}

export function deleteTrip(id) {
    const request = axios
        .delete(`/api/tripDelete?id=${id}`);

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: DELETE_TRIP,
                payload: response.data
            });
        });
    }
}

export function clearTrip() {
    return {
        type: CLEAR_TRIP,
        payload: {
            trip: null,
            updateTrip: false,
            postDeleted: false
        }
    }
}

export function clearTripWithReviewer() {
    return {
        type: CLEAR_TRIP_W_REVIEWER,
        payload: {
            trip: null,
            reviewer: null
        }
    };
}

export function loginUser({ email, password }) {
    const request = axios
        .post('/api/login', { email, password });

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: USER_LOGIN,
                payload: response.data
            });
        });
    }
}

export function resetPassword({ email }) {
    const request = axios
        .post('/api/resetPassword', { email });

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: RESET_PASSWORD,
                payload: response.data
            });
        });
    }
}

export function auth() {
    const request = axios.get('/api/auth');

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: USER_AUTH,
                payload: response.data
            });
        });
    }
}

export function addTrip(trip) {
    const request = axios.post('/api/trip', trip);

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: ADD_TRIP,
                payload: response.data
            });
        });
    }
}

export function clearNewTrip() {
    return {
        type: CLEAR_NEW_TRIP,
        payload: null
    };
}

export function getUsers() {
    const request = axios.get(`/api/users`);

    return (dispatch) => {
        return request.then((response) => {
            dispatch({
                type: GET_USERS,
                payload: response.data
            });
        });
    }
}

export function userRegister(user) {
    const request = axios.post(`/api/register`, user);

    return (dispatch) => {
        return request.then(({ data }) => {
            const response = {
                success: data.success
            };

            dispatch({
                type: USER_REGISTER,
                payload: response
            });
        })
    }
}