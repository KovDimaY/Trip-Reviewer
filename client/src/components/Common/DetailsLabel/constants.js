export const TYPES = {
  expences: 'EXPENCES',
  duration: 'DURATION',
  rating: 'RATING',
};

const colors = {
  very_good: '#00B300',
  good: '#7FBB17',
  ok: '#FEC22D',
  bad: '#F47048',
  very_bad: '#E91E63',
};

const limitsExpences = [100, 500, 1000, 5000];
const limitsDuration = [2, 7, 14, 31];
const limitsRating = [1, 2, 3, 4];

const getColorsInversed = (value, limits) => {
  switch (true) {
    case value > limits[3]:
      return colors.very_bad;
    case value > limits[2]:
      return colors.bad;
    case value > limits[1]:
      return colors.ok;
    case value > limits[0]:
      return colors.good;
    default:
      return colors.very_good;
  }
};

const getColorsDirect = (value, limits) => {
  switch (true) {
    case value > limits[3]:
      return colors.very_good;
    case value > limits[2]:
      return colors.good;
    case value > limits[1]:
      return colors.ok;
    case value > limits[0]:
      return colors.bad;
    default:
      return colors.very_bad;
  }
};

const getFunctionByType = {
  [TYPES.expences]: value => getColorsInversed(value, limitsExpences),
  [TYPES.duration]: value => getColorsDirect(value, limitsDuration),
  [TYPES.rating]: value => getColorsDirect(value, limitsRating),
};

export const getColorsByTypeAndValue = (type, value) => getFunctionByType[type](value);
