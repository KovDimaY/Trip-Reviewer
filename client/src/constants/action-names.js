export const GET_TRIPS = 'get-trips';
export const GET_TRIP_W_REVIEWER = 'get-trip-w-reviewer';
export const CLEAR_TRIP_W_REVIEWER = 'clear-trip-w-reviewer';
export const USER_LOGIN = 'user-login';
export const USER_AUTH = 'user-auth';