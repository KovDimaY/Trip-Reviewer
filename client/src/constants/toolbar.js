const toolbar = {
  options: ['inline', 'list', 'textAlign', 'link', 'emoji', 'image', 'remove', 'history'],
  inline: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['bold', 'italic', 'underline', 'strikethrough'],
  },
  list: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['unordered', 'ordered', 'indent', 'outdent'],
  },
  textAlign: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['left', 'center', 'right', 'justify'],
  },
  link: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    dropdownClassName: undefined,
    showOpenOptionOnHover: true,
    defaultTargetOption: '_self',
  },
  emoji: {
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    emojis: [
      '😀', '😁', '😂', '😃', '😉', '😋', '😎', '😍', '😗', '🤗', '🤔', '😣', '😫', '😴', '😌', '🤓',
      '😛', '😜', '😠', '😇', '😷', '😈', '👻', '😺', '😸', '😹', '😻', '😼', '😽', '🙀', '🙈',
      '🙉', '🙊', '👼', '👮', '🕵', '💂', '👳', '🎅', '👸', '👰', '👲', '🙍', '🙇', '🚶', '🏃', '💃',
      '⛷', '🏂', '🏌', '🏄', '🚣', '🏊', '⛹', '🏋', '🚴', '👫', '💪', '👈', '👉', '👉', '👆', '🖕',
      '👇', '🖖', '🤘', '🖐', '👌', '👍', '👎', '✊', '👊', '👏', '🙌', '🙏', '🐵', '🐶', '🐇', '🐥',
      '🐸', '🐌', '🐛', '🐜', '🐝', '🍉', '🍄', '🍔', '🍤', '🍨', '🍪', '🎂', '🍰', '🍾', '🍷', '🍸',
      '🍺', '🌍', '🚑', '⏰', '🌙', '🌝', '🌞', '⭐', '🌟', '🌠', '🌨', '🌩', '⛄', '🔥', '🎄', '🎈',
      '🎉', '🎊', '🎁', '🎗', '🏀', '🏈', '🎲', '🔇', '🔈', '📣', '🔔', '🎵', '🎷', '💰', '🖊', '📅',
      '✅', '❎', '💯',
    ],
  },
  image: {
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    urlEnabled: true,
    uploadEnabled: true,
    alignmentEnabled: true,
    uploadCallback: undefined,
    previewImage: false,
    inputAccept: 'image/gif,image/jpeg,image/jpg,image/png,image/svg',
    alt: { present: false, mandatory: false },
    defaultSize: {
      height: 'auto',
      width: 'auto',
    },
  },
  remove: { className: undefined, component: undefined },
  history: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['undo', 'redo'],
  },
};

export default toolbar;
