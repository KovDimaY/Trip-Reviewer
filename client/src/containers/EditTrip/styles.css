.edit-review-container {
    text-align: center;
}
.edit-review-container .form_element {
    margin-top: 20px;
}
.edit-review-container h2 {
    font-size: 33px;
    font-weight: 300;
    color: #02bcd4;
}
.edit-review-container input {
    font-size: 20px;
    border-radius: 3px;
    border: 1px solid #ebebeb;
    box-sizing: border-box;
    padding: 11px 31px;
    background: #f7f7f7;
    font-weight: 300;
    width: 80%;
}
.edit-review-container .label {
    display: block;
    text-align: left;
    margin-left: 10%;
    color: #02bcd4;
    font-size: 21px;
    margin-bottom: 3px;
}
.edit-review-container textarea {
    background: #f7f7f7;
    border: 1px solid #ebebeb;
    border-radius: 3px;
    width: 80%;
    padding: 20px;
    box-sizing: border-box;
    font-size: 14px;
}
.edit-review-container select {
    width: 80%;
    background: #f7f7f7;
    font-size: 20px;
    border: 1px solid #e1e1e1;
}
.edit-review-container button {
    font-size: 20px;
    padding: 10px 52px;
    background: #02bcd4;
    color: #fff;
    font-weight: 300;
    border-radius: 3px;
    border: none;
    margin-top: 50px;
    width: 80%;
}
.edit-review-container .conf_link {
    color: #8BC34A;
    font-weight: 600;
    margin-top: 20px;
}
.edit-review-container .error {
    color: #FF5722;
    margin-top: 10px;
    font-weight: 600;
}
.edit-review-container .delete_post {
    text-align: center;
    margin-top: 20px;
}
.edit-review-container .delete_post .button {
    font-size: 20px;
    padding: 10px 52px;
    background: #E91E63;
    color: #fff;
    font-weight: 300;
    border-radius: 3px;
    border: none;
    width: 80%;
    box-sizing: border-box;
    display: inline-block;
    margin-bottom: 50px;
}
.edit-review-container .editor-wrapper {
    width: 80%;
    margin-left: 10%;
    border: 1px solid #ebebeb;
    background-color: #f7f7f7;
}
.edit-review-container .editor-self {
    padding-left: 10px;
    padding-right: 10px; 
}