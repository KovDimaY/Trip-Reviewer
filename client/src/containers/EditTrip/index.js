import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import StarsRating from './../../components/StarsRating';
import { getTrip, updateTrip, clearTrip, deleteTrip } from '../../actions';

class EditTrip extends PureComponent {
    state = {
        formdata: {
            _id: this.props.match.params.id,
            title: '',
            author: '',
            review: '',
            duration: '',
            rating: 0,
            price: ''
        }
    };

    componentWillMount() {
        this.props.dispatch(getTrip(this.props.match.params.id));
    }

    componentWillReceiveProps(nextProps) {
        const { trip, updateTrip } = nextProps.trips

        if (updateTrip) {
            nextProps.history.push(`/trips/${trip._id}`);
        } else {
            this.setState({
                formdata: {
                    _id: trip._id,
                    title: trip.title,
                    author: trip.author,
                    review: trip.review,
                    duration: trip.duration,
                    rating: trip.rating,
                    price: trip.price
                }
            });
        }
    }

    componentWillUnmount() {
        this.props.dispatch(clearTrip());
    }

    handleInput = (event) => {
        const newFormdata = {
            ...this.state.formdata
        };
        const { value, name } = event.target;

        newFormdata[name] = value;

        this.setState({
            formdata: newFormdata
        });
    }

    handleRating = (rating) => {
        const newFormdata = {
            ...this.state.formdata
        };

        newFormdata.rating = rating;

        this.setState({
            formdata: newFormdata
        });
    }

    submitForm = (event) => {
        event.preventDefault();
        this.props.dispatch(updateTrip(this.state.formdata));
    }

    deletePost = () => {
        this.props.dispatch(deleteTrip(this.props.match.params.id));
    }

    goToReviews = () => {
        this.props.history.push('/user/user-reviews');
    }

    redirectUser = () => {
        setTimeout(this.goToReviews, 1000);
    }

    render() {
        const {
            title, author, review,
            duration, rating, price
        } = this.state.formdata;
        const { trips } = this.props;

        return (
            <div className="rl_container article">
                {
                    trips.postDeleted 
                    ? <div className="red_tag">
                            Post Deleted
                            {this.redirectUser()}
                        </div>
                    : null
                }
                <form onSubmit={this.submitForm}>
                    <h2>Edit review</h2>

                    <div className="form_element">
                        <input
                            type="text"
                            name="title"
                            placeholder="Enter title"
                            value={title}
                            onChange={this.handleInput}
                        />
                    </div>

                    <div className="form_element">
                        <input
                            type="text"
                            name="author"
                            placeholder="Enter author"
                            value={author}
                            onChange={this.handleInput}
                        />
                    </div>

                    <textarea
                        value={review}
                        name="review"
                        onChange={this.handleInput}
                    />

                    <div className="form_element">
                        <input
                            type="number"
                            name="duration"
                            placeholder="Enter duration"
                            value={duration}
                            onChange={this.handleInput}
                        />
                    </div>

                    <div className="form_element">
                        <StarsRating rating={rating} onChange={this.handleRating} />
                    </div>

                    <div className="form_element">
                        <input
                            type="number"
                            name="price"
                            placeholder="Enter Price"
                            value={price}
                            onChange={this.handleInput}
                        />
                    </div>

                    <button type="submit">Edit review</button>
                    <div className="delete_post">
                        <div className="button"
                            onClick={this.deletePost}
                        >
                            Delete review
                        </div>
                    </div>
                </form>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        trips: state.trips
    }
}

export default connect(mapStateToProps)(EditTrip);