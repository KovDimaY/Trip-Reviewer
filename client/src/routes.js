import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Home from './components/Home';
import User from './components/User';
import Logout from './components/Logout';
import Login from './containers/Login';
import Register from './containers/Register';
import AddTrip from './containers/AddTrip';
import EditTrip from './containers/EditTrip';
import TripView from './containers/TripView';
import UserReviews from './containers/UserReviews';

import Layout from './hoc/layout';
import Auth from './hoc/auth';

const Routes = () => {
    return (
        <Layout>
            <Switch>
                <Route path="/" exact component={Auth(Home, null)}/>
                <Route path="/login" exact component={Auth(Login, false)}/>
                <Route path="/user/logout" exact component={Auth(Logout, true)}/>
                <Route path="/user/register" exact component={Auth(Register, false)}/>
                <Route path="/user" exact component={Auth(User, true)}/>
                <Route path="/user/user-reviews" exact component={Auth(UserReviews, true)}/>
                <Route path="/user/edit-post/:id" exact component={Auth(EditTrip, true)}/>
                <Route path="/trip/add" exact component={Auth(AddTrip, true)}/>
                <Route path="/trips/:id" exact component={Auth(TripView, null)}/>
            </Switch>
        </Layout>
    );
};

export default Routes;