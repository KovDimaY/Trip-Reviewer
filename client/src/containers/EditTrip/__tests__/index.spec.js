import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
 
import EditTrip from './../../EditTrip';
import { GET_TRIP } from './../../../constants/action-names';
import { updateTrip, clearTrip, deleteTrip } from './../../../actions';
import * as routes from './../../../constants/routes';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
jest.mock('./../../../components/StarsRating', () => 'StarsRating');
jest.mock('./../../../actions', () => ({ 
    getTrip: jest.fn(() => ({
        type: GET_TRIP,
        payload: {}
    })),
    updateTrip: jest.fn(() => ({
        type: 'updateTrip'
    })),
    clearTrip: jest.fn(() => ({
        type: 'clearTrip'
    })),
    deleteTrip: jest.fn(() => ({
        type: 'deleteTrip'
    }))
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
    const store = mockStore(initialState);

    return (
        <EditTrip {...props} store={store} />
    );
};
 
describe('<EditTrip />', () => {
    it('should render component', () => {
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const tree = create(mockComponent(initialState, props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should render component with updateTrip', () => {
        const initialState = {
            trips: {
                updateTrip: true,
                trip: {
                    _id: '_id'
                }
            }
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const tree = create(mockComponent(initialState, props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should render component with postDeleted', () => {
        const initialState = {
            trips: {
                postDeleted: true
            }
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const tree = create(mockComponent(initialState, props)).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should change state when componentWillReceiveProps is called', () => {
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const nextProps = {
            trips: {
                trip: {
                    _id: 'trip._id',
                    title: 'trip.title',
                    author: 'trip.author',
                    review: 'trip.review',
                    duration: 'trip.duration',
                    rating: 'trip.rating',
                    price: 'trip.price'
                }
            }
        };
        
        const instance = shallow(mockComponent(initialState, props)).dive().instance();
    
        instance.componentWillReceiveProps(nextProps);
   
        expect(instance.state.formdata).toEqual(nextProps.trips.trip);
    });

    it('should redirect when componentWillReceiveProps is called with updateTrip', () => {
        const push = jest.fn();
        const tripId = 'test';
        const path = `${routes.TRIPS}/${tripId}`;
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const nextProps = {
            trips: {
                trip: {
                    _id: tripId
                },
                updateTrip: true
            },
            history: {
                push
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();

        instance.componentWillReceiveProps(nextProps);

        expect(push).toHaveBeenCalledWith(path);
    });

    it('should dispatch clearTrip when componentWillUnmount is called', () => {
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();
    
        instance.componentWillUnmount();

        expect(clearTrip).toHaveBeenCalled();
    });

    it('should dispatch updateTrip when submitForm is called', () => {
        const preventDefault = jest.fn();
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const state = {
            author: "author",
            duration: "duration",
            ownerId: "id",
            price: "price",
            rating: "rating",
            review: "review",
            title: "title"
        };
        const initialState = {
            trips: {}
        };
        const event = { preventDefault };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();
        instance.setState({ formdata: state });

        instance.submitForm(event);

        expect(preventDefault).toHaveBeenCalled();    
        expect(updateTrip).toHaveBeenCalledWith(state);
    });

    it('should dispatch deleteTrip when deletePost is called', () => {
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();

        instance.deletePost();

        expect(deleteTrip).toHaveBeenCalledWith('id');
    });

    it('handleInput should change state correctly', () => {
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const event = {
            target: {
                value: 'test',
                name: 'title'
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();
    
        instance.handleInput(event);
    
        expect(instance.state.formdata.title).toEqual('test');
    });

    it('handleRating should change state correctly', () => {
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            }
        };
        const rating = 3;

        const instance = shallow(mockComponent(initialState, props)).dive().instance();

        instance.handleRating(rating);

        expect(instance.state.formdata.rating).toEqual(rating);
    });

    it('goToReviews should redirect correctly', () => {
        const push = jest.fn();
        const path = routes.USER_REVIEWS;
        const initialState = {
            trips: {}
        };
        const props = {
            match: {
                params: {
                    id: 'id'
                }
            },
            history: {
                push
            }
        };

        const instance = shallow(mockComponent(initialState, props)).dive().instance();

        instance.goToReviews();

        expect(push).toHaveBeenCalledWith(path);
    });


});