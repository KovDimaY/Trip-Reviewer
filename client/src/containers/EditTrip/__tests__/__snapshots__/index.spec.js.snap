// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditTrip /> should render component 1`] = `
<div
  className="rl_container article"
>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Edit review
    </h2>
    <div
      className="form_element"
    >
      <input
        name="title"
        onChange={[Function]}
        placeholder="Enter title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <input
        name="author"
        onChange={[Function]}
        placeholder="Enter author"
        type="text"
        value=""
      />
    </div>
    <textarea
      name="review"
      onChange={[Function]}
      value=""
    />
    <div
      className="form_element"
    >
      <input
        name="duration"
        onChange={[Function]}
        placeholder="Enter duration"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <select
        name="rating"
        onChange={[Function]}
        value=""
      >
        <option
          val="1"
        >
          1
        </option>
        <option
          val="2"
        >
          2
        </option>
        <option
          val="3"
        >
          3
        </option>
        <option
          val="4"
        >
          4
        </option>
        <option
          val="5"
        >
          5
        </option>
      </select>
    </div>
    <div
      className="form_element"
    >
      <input
        name="price"
        onChange={[Function]}
        placeholder="Enter Price"
        type="number"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Edit review
    </button>
    <div
      className="delete_post"
    >
      <div
        className="button"
        onClick={[Function]}
      >
        Delete review
      </div>
    </div>
  </form>
</div>
`;

exports[`<EditTrip /> should render component with postDeleted 1`] = `
<div
  className="rl_container article"
>
  <div
    className="red_tag"
  >
    Post Deleted
  </div>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Edit review
    </h2>
    <div
      className="form_element"
    >
      <input
        name="title"
        onChange={[Function]}
        placeholder="Enter title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <input
        name="author"
        onChange={[Function]}
        placeholder="Enter author"
        type="text"
        value=""
      />
    </div>
    <textarea
      name="review"
      onChange={[Function]}
      value=""
    />
    <div
      className="form_element"
    >
      <input
        name="duration"
        onChange={[Function]}
        placeholder="Enter duration"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <select
        name="rating"
        onChange={[Function]}
        value=""
      >
        <option
          val="1"
        >
          1
        </option>
        <option
          val="2"
        >
          2
        </option>
        <option
          val="3"
        >
          3
        </option>
        <option
          val="4"
        >
          4
        </option>
        <option
          val="5"
        >
          5
        </option>
      </select>
    </div>
    <div
      className="form_element"
    >
      <input
        name="price"
        onChange={[Function]}
        placeholder="Enter Price"
        type="number"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Edit review
    </button>
    <div
      className="delete_post"
    >
      <div
        className="button"
        onClick={[Function]}
      >
        Delete review
      </div>
    </div>
  </form>
</div>
`;

exports[`<EditTrip /> should render component with updateTrip 1`] = `
<div
  className="rl_container article"
>
  <div
    className="edit_confirm"
  >
    Post updated, 
    <Link
      to="/trips/_id"
    >
      Click here to see your post
    </Link>
  </div>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Edit review
    </h2>
    <div
      className="form_element"
    >
      <input
        name="title"
        onChange={[Function]}
        placeholder="Enter title"
        type="text"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <input
        name="author"
        onChange={[Function]}
        placeholder="Enter author"
        type="text"
        value=""
      />
    </div>
    <textarea
      name="review"
      onChange={[Function]}
      value=""
    />
    <div
      className="form_element"
    >
      <input
        name="duration"
        onChange={[Function]}
        placeholder="Enter duration"
        type="number"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <select
        name="rating"
        onChange={[Function]}
        value=""
      >
        <option
          val="1"
        >
          1
        </option>
        <option
          val="2"
        >
          2
        </option>
        <option
          val="3"
        >
          3
        </option>
        <option
          val="4"
        >
          4
        </option>
        <option
          val="5"
        >
          5
        </option>
      </select>
    </div>
    <div
      className="form_element"
    >
      <input
        name="price"
        onChange={[Function]}
        placeholder="Enter Price"
        type="number"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Edit review
    </button>
    <div
      className="delete_post"
    >
      <div
        className="button"
        onClick={[Function]}
      >
        Delete review
      </div>
    </div>
  </form>
</div>
`;
